wait_for/__init__.py

```diff
@@ -177,7 +179,11 @@ def wait_for(func, func_args=[], func_kwargs={}, logger=None, **kwargs):
         logger.error("Couldn't complete {} at {}:{} in time, took {:0.2f}, {} tries".format(message,
             filename, line_no, t_delta, tries))
         logger.error('The last result of the call was: {}'.format(out))
-        raise TimedOutError("Could not do {} at {}:{} in time".format(message, filename, line_no))
+        if timeout_func:
+            return timeout_func()
```

Comment:
I'm concerned that we have a ```WaitForResult``` that we "usually" return and that in this case we don't. It could be confusing. Is there a need to return the result of ```timeout_func()```. The other way to sort this is to have an "error_code" in the named tuple for the ```WaitForResult```. So that we know the output is not from the wait_for but from the timeout.

Reply:
At the moment I can't figure out when the result of timeout_func can be useful, hence I think it will be better to return nothing.

Reply:
I think we are adding perhaps too much complexity into this, I'm not sure that this adds much extra over the try: except pattern, which arguably I think is the right way to do it.